### ISSUE TITLE
Repeatedly typing space in extension search box causes hang

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/vscode#55426]
Testing #55386

Issue Type: <b>Bug</b>

1. Open extension search box.
1. Repeatedly type `space` every half second or so

**Bug**
After five or more spaces, the entire VS Code UI starts hanging for a few seconds

VS Code version: Code - Insiders 1.26.0-insider (0707dd4940b3fdbea8339595de478f354f6dda83, 2018-07-31T05:12:08.688Z)
OS version: Darwin x64 17.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz (8 x 2200)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 5|
|Memory (System)|16.00GB (0.14GB free)|
|Process Argv|/Applications/Visual Studio Code - Insiders.app/Contents/MacOS/Electron -psn_0_9410809|
|Screen Reader|no|
|VM|17%|

</details>

**Comment by JacksonKearl** [comment]
Are you able to reproduce this reliably? I've never gotten it on Windows but I did manage once on macOS, even then it only stalled the extensions viewlet UI, rest was fine. Its just pressing space as the search triggers right?

**Comment by ramya-rao-a** [comment]
@JacksonKearl Can we not skip the search if newValue.trim() === oldValue.trim() ?

**Comment by JacksonKearl** [comment]
We can do that. But what if they hit 'a' a bunch? @mjbvz can you repro this in stable?

**Comment by mjbvz** [comment]
No, I didn’t see this in stable when typing spaces. I also don’t remember being able to reproduce this with other characters, such as typing “a” many times

**Comment by JacksonKearl** [comment]
Then we can probably get away with just debouncing the trimmed value

**Comment by ramya-rao-a** [comment]
@JacksonKearl Let's do that to begin with
